# Copyright 2026 The OpenXLA Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
Silence match and rewrite failures for the reshape memdesc pattern.
Previously, failure to match due to type inference failure would log to
stderr, which is unexpected for a match failure.

This is caused by the `verify_nvmma_encoding` patch, and can therefore only be
upstreamed with that patch. See context in b/504826033.

--- a/lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp
+++ b/lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp
@@ -161,7 +161,7 @@ public:
     // `allocType` after a reshape.
     MemDescType innerTy;
     if (failed(MemDescReshapeOp::inferReturnTypes(
-            getContext(), allocOp.getLoc(), allocType, srcShape, innerTy)))
+            getContext(), /*loc=*/std::nullopt, allocType, srcShape, innerTy)))
       return failure();
 
     // For now don't apply the transformation if the new encoding is not an

